### PR TITLE
Project Staffing | Edit and add project personnel

### PR DESCRIPTION
**Pull request by mddilley**
Closes https://github.com/cityofaustin/atd-data-tech/issues/4913

This PR adds the add and edit personnel features to the team table in the Project Summary Team tab. Using this table in the New Project workflow and also deleting (making them active/inactive) users will come in further PRs.

This can be tested locally with the seeded project with one team member. https://localhost:3000/moped/projects/1?tab=team

### Original
Screen Shot 2021-02-11 at 2 19 53 PM

### Add personnel
Screen Shot 2021-02-11 at 2 20 14 PM

### Edit personnel
Screen Shot 2021-02-11 at 2 20 38 PM

### CODE REVIEW

File: moped-editor/src/views/projects/projectView/ProjectTeam.js [hunk 1]
```diff
@@ -1,103 +1,17 @@
 import React from "react";
-import { useQuery } from "@apollo/client";
+import ProjectTeamTable from "./ProjectTeamTable";
```
Comment (sergiogcx) [inline moped-editor/src/views/projects/projectView/ProjectTeam.js:2]: Happy to see this offloaded :)

File: moped-editor/src/views/projects/projectView/ProjectTeamTable.js [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+  const filterObjectByKeys = (obj, keys) =>
+    Object.keys(obj)
+      .filter(key => !keys.includes(key))
+      .reduce(
+        (acc, key) => ({
+          ...acc,
+          [key]: obj[key],
+        }),
+        {}
+      );
```
Comment (sergiogcx) [inline moped-editor/src/views/projects/projectView/ProjectTeamTable.js:98]: Great commentary and great code! :) I think it's the first time I see reduce in moped 💯

File: moped-editor/src/views/projects/projectView/ProjectTeamTable.js [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+  projectState,
+  setProjectState,
+  projectId = null,
+}) => {
+  const isNewProject = projectId === null;
+
+  const { loading, error, data, refetch } = useQuery(TEAM_QUERY, {
+    variables: { projectId },
+    fetchPolicy: "no-cache",
+  });
```
Comment (sergiogcx) [inline moped-editor/src/views/projects/projectView/ProjectTeamTable.js:25]: I thought cache was no longer a problem? not a issue whatsoever, just trying to recall what the status is

File: moped-editor/src/views/projects/projectView/ProjectTeamTable.js [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+          ...acc,
+          [key]: obj[key],
+        }),
+        {}
+      );
+
+  /**
+   * Column configuration for <MaterialTable>
+   */
+  const columns = [
```
Comment (sergiogcx) [inline moped-editor/src/views/projects/projectView/ProjectTeamTable.js:103]: This whole config file is very clever! 👍

**Review by sergiogcx — APPROVED** [review]
Brilliant! Would not change one thing! :)

🚢

File: moped-editor/src/views/projects/projectView/ProjectTeamTable.js (hunk 3 above)
Comment (mddilley) [reply moped-editor/src/views/projects/projectView/ProjectTeamTable.js:26]: Yep, cache is not an issue for Apollo anymore, but Material Table breaks without this fetch policy. I'll take a look in the last PR of this series. Thanks for the heads up!

**Pull request merged**